File: app/src/main/java/com/storyshu/storyshu/adapter/MyStoryAdapter.java
package com.storyshu.storyshu.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.storyshu.storyshu.R;
import com.storyshu.storyshu.info.CardInfo;
import com.storyshu.storyshu.utils.time.TimeUtils;
import com.storyshu.storyshu.widget.ClickButton;
import com.storyshu.storyshu.widget.imageview.AvatarImageView;

import java.util.List;

/**
 * 我的故事卡片的适配器
 * Created by bear on 2017/5/23.
 */

public class MyStoryAdapter extends RecyclerView.Adapter<MyStoryAdapter.ViewHold> {
    private Context mContext;

    private List<CardInfo> myStoryList;

    private OnStoryCardClickListener onCardClickedListener;

    public void setOnCardClickedListener(OnStoryCardClickListener onCardClickedListener) {
        this.onCardClickedListener = onCardClickedListener;
    }

    public interface OnStoryCardClickListener {
        void onClick(int position);
    }

    public MyStoryAdapter(Context context, List<CardInfo> cardInfoList) {
        this.myStoryList = cardInfoList;
        this.mContext = context;
    }

    @Override
    public ViewHold onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.my_story_item_layout, parent, false);
        ViewHold viewHolder = new ViewHold(view);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(ViewHold holder, int position) {
        CardInfo cardInfo = myStoryList.get(position);

        if (position > 0) {
            if (TimeUtils.getDate(cardInfo.getCreateTime()).equals(
                    TimeUtils.getDate(myStoryList.get(position - 1).getCreateTime()))) {

                holder.day.setVisibility(View.GONE);
                holder.month.setVisibility(View.GONE);
            } else {

                holder.day.setVisibility(View.VISIBLE);
                holder.month.setVisibility(View.VISIBLE);

                holder.day.setText(TimeUtils.getDay(cardInfo.getCreateTime()));
                holder.month.setText(mContext.getString(R.string.unit_month, TimeUtils.getMonth(cardInfo.getCreateTime())));
            }

        } else {

            holder.day.setVisibility(View.VISIBLE);
            holder.month.setVisibility(View.VISIBLE);

            holder.day.setText(TimeUtils.getDay(cardInfo.getCreateTime()));
            holder.month.setText(mContext.getString(R.string.unit_month, TimeUtils.getMonth(cardInfo.getCreateTime())));
        }

        Glide.with(mContext).load(cardInfo.getUserInfo().getAvatar()).into(holder.avatar);
        holder.nickName.setText(cardInfo.getUserInfo().getNickname());

        holder.content.setText(cardInfo.getContent());
        holder.location.setText(cardInfo.getLocationTitle());
        holder.destoryTime.setText(TimeUtils.convertDestroyTime(mContext, cardInfo.getDestroyTime()));

        holder.like.setNum(cardInfo.getLikeNum());
        holder.oppose.setNum(cardInfo.getOpposeNum());
        holder.comment.setNum(cardInfo.getCommentNum());

        if (TextUtils.isEmpty(cardInfo.getCover())) {
            holder.cover.setVisibility(View.GONE);
        } else {
            holder.cover.setVisibility(View.VISIBLE);
            Glide.with(mContext).load(cardInfo.getCover()).into(holder.cover);
        }
    }

    @Override
    public int getItemCount() {
        return myStoryList == null ? 0 : myStoryList.size();
    }

    class ViewHold extends RecyclerView.ViewHolder implements View.OnClickListener {
        private TextView day;
        private TextView month;

        private AvatarImageView avatar; //用户头像
        private TextView nickName; //用户昵称
        private TextView content;
        private TextView location;
        private ImageView cover;
        private TextView destoryTime;

        private ClickButton like; //点赞
        private ClickButton oppose;  //喝倒彩
        private ClickButton comment;  //喝倒彩

        public ViewHold(View itemView) {
            super(itemView);

            itemView.setOnClickListener(this);
            day = (TextView) itemView.findViewById(R.id.day);
            month = (TextView) itemView.findViewById(R.id.month);
            avatar = (AvatarImageView) itemView.findViewById(R.id.avatar);
            nickName = (TextView) itemView.findViewById(R.id.nickname);
            like = (ClickButton) itemView.findViewById(R.id.like);
            oppose = (ClickButton) itemView.findViewById(R.id.oppose);
            comment = (ClickButton) itemView.findViewById(R.id.comment);
            destoryTime = (TextView) itemView.findViewById(R.id.destroy_time);
            content = (TextView) itemView.findViewById(R.id.content);
            cover = (ImageView) itemView.findViewById(R.id.cover_pic);
            location = (TextView) itemView.findViewById(R.id.location);
        }

        @Override
        public void onClick(View v) {
            if (onCardClickedListener != null)
                onCardClickedListener.onClick(getLayoutPosition());
        }
    }
}

File: app/src/main/java/com/storyshu/storyshu/adapter/MyStoryAdapter.kt
package com.storyshu.storyshu.adapter

import android.text.TextUtils
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.storyshu.storyshu.R
import com.storyshu.storyshu.info.CardInfo
import com.storyshu.storyshu.utils.time.TimeUtils
import com.storyshu.storyshu.widget.ClickButton
import com.storyshu.storyshu.widget.imageview.AvatarImageView

/**
 * 我的故事卡片的适配器
 * Created by bear on 2017/5/23.
 */

class MyStoryAdapter(myStoryList: List<CardInfo>?) :
        BaseQuickAdapter<CardInfo, MyStoryAdapter.ViewHold>(R.layout.my_story_item_layout, myStoryList) {

    interface OnStoryCardClickListener {
        fun onClick(position: Int)
    }


    override fun convert(helper: ViewHold?, item: CardInfo?) {
        if (helper!!.layoutPosition > 0) {
            if (TimeUtils.getDate(item?.createTime) == TimeUtils.getDate(getItem(helper.layoutPosition-1).createTime)) {

                helper?.day.visibility = View.GONE
                helper?.month.visibility = View.GONE
            } else {

                helper?.day.visibility = View.VISIBLE
                helper?.month.visibility = View.VISIBLE

                helper?.day.text = TimeUtils.getDay(item?.createTime)
                helper?.month.text = mContext.getString(R.string.unit_month, TimeUtils.getMonth(item?.createTime))
            }

        } else {

            helper?.day.visibility = View.VISIBLE
            helper?.month.visibility = View.VISIBLE

            helper?.day.text = TimeUtils.getDay(item?.createTime)
            helper?.month.text = mContext.getString(R.string.unit_month, TimeUtils.getMonth(item?.createTime))
        }

        Glide.with(mContext).load(item?.userInfo?.avatar).into(helper?.avatar)
        helper?.nickName.text = item?.userInfo?.nickname

        helper?.location.text = item?.locationTitle
        helper?.destoryTime.text = TimeUtils.convertDestroyTime(mContext, item?.destroyTime)

        helper?.like.num = item?.likeNum ?:0
        helper?.oppose.num = item?.opposeNum?:0
        helper?.comment.num = item?.commentNum?:0

        if (TextUtils.isEmpty(item?.cover)) {
            helper?.cover.visibility = View.GONE
        } else {
            helper?.cover.visibility = View.VISIBLE
            Glide.with(mContext).load(item?.cover).into(helper?.cover)
        }

        when (item?.storyType) {
            CardInfo.STORY -> {
                helper?.content.setTextColor(mContext.resources.getColor(R.color.colorBlack))
                helper?.content.text = item?.content
            }

            CardInfo.ARTICLE -> {
                helper?.content.setTextColor(mContext.resources.getColor(R.color.colorRedLight))
                helper?.content.text = mContext.getString(R.string.story_type_article, item?.title)
            }
        }
    }


    inner class ViewHold(itemView: View) : BaseViewHolder(itemView) {
        val day: TextView
        val month: TextView

        val avatar: AvatarImageView //用户头像
        val nickName: TextView //用户昵称
        val content: TextView
        val location: TextView
        val cover: ImageView
        val destoryTime: TextView

        val like: ClickButton //点赞
        val oppose: ClickButton  //喝倒彩
        val comment: ClickButton  //喝倒彩

        init {
            day = itemView.findViewById(R.id.day) as TextView
            month = itemView.findViewById(R.id.month) as TextView
            avatar = itemView.findViewById(R.id.avatar) as AvatarImageView
            nickName = itemView.findViewById(R.id.nickname) as TextView
            like = itemView.findViewById(R.id.like) as ClickButton
            oppose = itemView.findViewById(R.id.oppose) as ClickButton
            comment = itemView.findViewById(R.id.comment) as ClickButton
            destoryTime = itemView.findViewById(R.id.destroy_time) as TextView
            content = itemView.findViewById(R.id.content) as TextView
            cover = itemView.findViewById(R.id.cover_pic) as ImageView
            location = itemView.findViewById(R.id.location) as TextView
        }

    }
}
